sympy/solvers/polysys.py

```diff
@@ -429,3 +432,567 @@ def solve_triangulated(polys, *gens, **args):
+    >>> from sympy.solvers.polysys import subresultant_polynomials
+
+    >>> subresultant_polynomials(x**2, x, x)
+    [0, x]
+    """
+    # ensure deg(f) >= deg(g)
+    if degree(f, mvar) < degree(g, mvar):
+        f, g = g, f
+
+
+    prs = subresultants(f, g, mvar)
+    if len(prs) <=1:
+        return []
+
+    subres_polys = [0] * (degree(g, mvar) + 1)
+
+    for i in reversed(range(2, len(prs))):
+
+        subres_polys[ degree(prs[i-1], mvar) -1 ] = prs[i]
+
+        if degree(prs[i], mvar) < degree(prs[i-1], mvar) - 1:
+            degree_jump = degree(prs[i-1], mvar) - degree(prs[i], mvar) - 1
+            subres_polys[ degree(prs[i], mvar) ]  =\
+                prs[i] * LC(prs[i], mvar) ** degree_jump
+
+    # get last one
+    subres_polys[-1] = prs[1] * LC(g, mvar) ** (degree(f, mvar) - degree(g, mvar) - 1)
+
+    # try to expand to simplify
+    for i in range(len(subres_polys)):
+        try:
+            subres_polys[i] = expand(subres_polys[i])
+        except Exception:
+            continue
+
+    return subres_polys
+
+
+def subresultant_coefficients(f, g, mvar):
+    """
+    Computes the principal subresultant coefficients (PSC). Given the
+    subresultant polynomials, in increasing degree, the ith PSC is
+    the coefficient of mvar^i in the ith subresultant.
+
+    Parameters
+    ==========
+
+    f: Expr or Poly
+        A polynomial
+    g: Expr or Poly
+        A polynomial
+    mvar: a generator
+        The "main variable".
+        Polynomials are treated as univariate in the mvar.
+
+
+    Returns
+    =======
+
+    List
+        The principal subresultant coefficients (PSC) of f and g.
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x
+    >>> from sympy.solvers.polysys import subresultant_coefficients
+
+    >>> subresultant_coefficients(x**2, x, x)
+    [0, 1]
+    """
+    subres_polys = subresultant_polynomials(f, g, mvar)
+
+    subres_coeffs = []
+
+    for i in range(len(subres_polys)):
+        curr_coeff = Poly(subres_polys[i], mvar).nth(i)
+        subres_coeffs.append(curr_coeff)
+
+    return subres_coeffs
+
+
+# gets real roots, numeric if not algebraic
+# returns them sorted
+# not a public function!
+def get_nice_roots(poly):
+
+    # its a constant
+    try:
+        if poly.is_number:
+            return []
+    except Exception:
+        return []
+
+    factors = factor_list(poly)[1] # the 0th is just a number
+
+    roots = set()
+    # get the roots of the factors that are polynomials
+    for factor in factors:
+        curr_factor = factor[0]
+
+        if curr_factor.is_number:
+            continue
+
+        try:
+            new_roots = real_roots(curr_factor)
+            for i, r in enumerate(new_roots):
+                # want to avoid CRootOf
+                if r.has(ComplexRootOf):
+                    new_roots[i] = r.evalf()
+        except NotImplementedError:
+            new_roots = [root for root in nroots(curr_factor) if root.is_real]
```

Comment:
I don't think we can get away with using numeric approximations here. If we are to handle the 0D cells then we need to do it exactly. I assume that is why the Eq case here fails:
```python
In [11]: solve_poly_system_cad([x**5 - x + 1 > 0], [x])
Out[11]: [{x: -0.167303978261419}]

In [12]: solve_poly_system_cad([x**5 - x + 1 < 0], [x])
Out[12]: [{x: -2.16730397826142}]

In [13]: solve_poly_system_cad([Eq(x**5 - x + 1, 0)], [x])
Out[13]: []
```
Also there is no reason why the outputs cannot be exact rational numbers for the inequality cases. In this particular example they could even be integers.

I think we really need the implementation to be based entirely on rational numbers and RootOf rather than radicals and floats. The `nroots` function is not reliable in general and neither is replacing RootOf with floats.